source/common/http/filter_manager.h

```diff
@@ -518,6 +518,11 @@ class FilterManagerCallbacks {
    * Returns a handle to the downstream callbacks, if available.
    */
   virtual OptRef<DownstreamStreamFilterCallbacks> downstreamCallbacks() { return {}; }
+  /**
+   * Returns if close from the upstream is to be handled with half-close semantics.
+   * This is used for HTTP/1.1 codec.
+   */
+  virtual bool isHalfCloseEnabled() PURE;
```

Comment:
sorry but is this comment accurate?  it looks like the plumbing here sets half close enabled when we create an upstream request regardless of if the upstream is HTTP/1.1, though we only actually set the half close for HTTP/1.1.  I think for the filter manager this is really less about half close and more about if we want to send data upstream after downstrea m is complete, right?